core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -280,8 +284,17 @@ public synchronized void addMutation(TableId table, Mutation m)
     mutations.addMutation(table, m);
     totalAdded++;
 
-    if (mutations.getMemoryUsed() >= maxMem / 2) {
-      startProcessing();
+    if (mutations.getMemoryUsed() >= maxMem / MAX_MEMORY_DIVISOR) {
+      if (!somethingFailed) {
+        startProcessing();
+        consecutiveFailedProcessCalls.set(0);
+      } else
+        consecutiveFailedProcessCalls.getAndIncrement();
```

Comment:
Could you not merge this and the conditional for >= 5000 ? if nothing failed then you'd never hit the conditional anyway.

Reply:
@phrocker,  I am not sure what you want me to do?    Do you think there is anything of value in this pull request? @keith-turner had reservations about it.  I created it so I thought it would help.

Reply:
Just pushed what I think is the change you wanted @phrocker .